app/src/main/java/com/stevesoltys/seedvault/settings/SettingsManager.kt

```diff
@@ -27,48 +30,67 @@ class SettingsManager(context: Context) {
 
     private var isStorageChanging: AtomicBoolean = AtomicBoolean(false)
 
-    private val blacklistedApps: HashSet<String> by lazy {
-        prefs.getStringSet(PREF_KEY_BACKUP_APP_BLACKLIST, emptySet()).toHashSet()
+    /**
+     * This gets accessed by non-UI threads when saving with [PreferenceManager]
+     * and when [isBackupEnabled] is called during a backup run.
+     * Therefore, it is implemented with a thread-safe [ConcurrentSkipListSet].
+     */
+    private val blacklistedApps: MutableSet<String> by lazy {
+        ConcurrentSkipListSet(prefs.getStringSet(PREF_KEY_BACKUP_APP_BLACKLIST, emptySet()))
```

Comment:
This hopefully fixes the non-persistent app opt-out blacklist. Please test!

Reply:
Are you unable to reproduce the issue?